modules/end-to-end/pages/building-tekton-tasks.adoc

```diff
@@ -0,0 +1,298 @@
+
+This document provides a step-by-step guide on how to onboard your Tekton tasks to {ProductName}, enabling their integration into your build pipelines.
+
+
+== Prerequisites
+
+Before you begin, ensure you have the following:
+
+* A Github repository containing your Tekton tasks with a catalog structure, containing all the additional files (README, TROUBLESHOOTING, etc…)
+
+----
+tasks/
+└── echo/
+    ├── 0.1/
+    │   ├── echo.yaml
+    │   ├── README.md
+    │   ├── USAGE.MD
+    │   └── TROUBLESHOOT.md
+    └── 0.2/
+        ├── echo.yaml
+        └── README.md
+----
+
+* {ProductName} Github application installed in the repository xref:installing:github-app.adoc[GitHub App]
+
+* A quay organization to be used with on-pr and push pipelines.
+
+* Public repositories within the quay organization to push the bundles to when using the on-pr and push pipelines.
+
+* A quay organization for the released bundles.
+
+* A public repository within the release organization for the trusted tasks list.
+
+* Access to the {ProductName} environment.
+
+* Necessary permissions to create and manage components in {ProductName}.
+
+
+== Onboarding Steps
+
+The onboarding of tasks to {ProductName} is similar to onboarding other applications/components to {ProductName}, with the exception of not having a Dockerfile (we are not building images).
+
+
+=== Application, Components and ImageRepository
+
+Tekton tasks are built as standard OCI artifacts in {ProductName}.
+
+The xref:building:/creating.adoc[Creating applications and components] page can be referenced for more information on how to create an Application for all tasks and a Component for each task within this Application, as well as an ImageRepository resource to be used within the quay organization.
+
+
+[NOTE] 
+====
+* Each task and task version in the application should be represented by a component, each component should contain the pipeline in use (*tekton-bundle-builder-oci-ta*) and the context to the task within the GH repository (*tasks/echo/0.1* for example).
+* The Component's naming convention is <task_name>-v<task_version> (echo-v01)
+====
+
+Once these resources are added, a PR with *on-pr* and *push* pipelineruns will be sent to your repository.
+
+[NOTE] 
+====
+* The *on-pr* pipelinerun will be created with the context we  have defined for the component and with *on-cel-expression* to run the pipelinerun only in case a change was detected in the component, for example:
+
+[source, yaml]
+----
+pipelinesascode.tekton.dev/on-cel-expression: |
+    event == "pull_request"
+    && target_branch == "main"
+    && (
+    "tasks/echo/0.1/***".pathChanged() || ".tekton/echo-v01-pull-request.yaml".pathChanged()
+    )
+----
+However, when *branch protection* is enabled in your GH repository it might prevent the pipelinerun to run when needed for the branch protection, in this case updating the *on-cel-expression* in the on-pr pipelinerun will solve the issue.
```

Comment:
> your GH repository it might prevent the pipelinerun to run when needed for the branch protection

I don't think I understand what you are saying here.

Reply:
I am referring to a case where we have a branch protection rule to run specific tests/pipelines.
There might be cases where the rule requires a specific pipeline to run but because of the cel expression it will not be triggered.
In this case the specific pipeline will be pending and will not run.

Reply:
Yes, I will add an example